firmware/quadruna/BMS/src/tasks.c

```diff
@@ -274,13 +276,15 @@ void tasks_init(void)
     // Configure and initialize SEGGER SystemView.
     SEGGER_SYSVIEW_Conf();
     LOG_INFO("BMS reset!");
+    can.can                   = &hfdcan1;
+    can.canMsgRecievecallback = NULL;
```

Comment:
If we want to make the file of `CanHandle` constant, we have to let the user of the `hw_can` give a right  `can_msg_received_callback` instead of `NULL`. So in `hw_can` and `hw_fdcan`, we can't have something like this.
```
    if (!handle->can_msg_received_callback)
        handle->can_msg_received_callback = io_can_msgReceivedCallback;
```
But I agree that making the field constant is much safer.